plugins/voxeldownsize/filter/VoxelDownsizeFilter.cpp

```diff
@@ -125,19 +127,38 @@ bool VoxelDownsizeFilter::insert(int gx, int gy, int gz)
 {
     if (m_populatedVoxels.size() > m_batchSize)
     {
-        std::set<std::tuple<int, int, int>> tempMap = m_populatedVoxels;
-        leveldb::WriteBatch batch;
-        for (auto itr=tempMap.begin(); itr!=tempMap.end(); ++itr)
+        std::set<std::tuple<int, int, int>> tempMap;
+        std::swap(tempMap, m_populatedVoxels);
+        auto temp_ldb = m_ldb;
+        auto chunkSize = m_ldbSyncChunkSize;
+        m_pool->add([chunkSize, temp_ldb, tempMap]()
         {
-            auto t=*itr;
-            auto val = std::to_string(std::get<0>(t)) +
-                       std::to_string(std::get<1>(t)) +
-                       std::to_string(std::get<2>(t));
-            batch.Put(val,val);
-        }
-        auto res = m_ldb->Write(leveldb::WriteOptions(), &batch).ok();
-        assert(res);
-        m_populatedVoxels.clear();
+            Pool localPool(100, 10);
```

Comment:
What's the logic for this double multi threading?

Reply:
Here, I want to spawn a thread to write to leveldb, which will spawn multiple child threads with small chunks of original set. I know this is a bit complex, But it is really fast.